### PR TITLE
[autodiscover] - Providing config option to disable Kubeadm config api requests

### PR DESCRIPTION
**Pull request by gizas**
Related changes needed for https://github.com/elastic/beats/pull/40086

### CODE REVIEW

File: kubernetes/metadata/metadata.go [hunk 1]
```diff
@@ -134,19 +133,23 @@ func GetKubernetesClusterIdentifier(cfg *config.C, client k8sclient.Interface) (
...
 
-func getClusterInfoFromKubeadmConfigMap(client k8sclient.Interface) (ClusterInfo, error) {
+func getClusterInfoFromKubeadmConfigMap(client k8sclient.Interface, kubeadm bool) (ClusterInfo, error) {
 	clusterInfo := ClusterInfo{}
 	if client == nil {
 		return clusterInfo, fmt.Errorf("unable to get cluster identifiers from kubeadm-config")
 	}
+	if kubeadm {
+		return clusterInfo, fmt.Errorf("unable to get cluster identifiers from kubeadm-config because conf_kubeadm access has been disabled")
+	}
```
Comment (AndersonQ) [inline kubernetes/metadata/metadata.go:152]: One question, will this be reported as an error in the logs? It seems to me if the user disabled that, it should not be reported as an error.

Comment (gizas) [reply kubernetes/metadata/metadata.go:152]: Good point. I will change it to Debug

Comment (gizas) [reply kubernetes/metadata/metadata.go]: Reverted to nil

File: kubernetes/metadata/config.go [hunk 2]
```diff
@@ -23,8 +23,8 @@ import (
 
 // Config declares supported configuration for metadata generation
 type Config struct {
-	KubeConfig string `config:"kube_config"`
-
+	KubeConfig         string   `config:"kube_config"`
+	KubeAdm            bool     `config:"disable_kubeadm"`
```
Comment (tetianakravchenko) [inline kubernetes/metadata/config.go:27]:
```suggestion
	UseKubeAdm            bool     `config:"use_kubeadm"`
```
I would use affirmative name instead, so `true` means enabled; `false` - disabled, not vice versa
imagine checking in code `c.KubeAdm = false` would give an impression that it is disabled, it introduce obscurity
default value should be changed in this case

Comment (gizas) [reply kubernetes/metadata/config.go:27]: You are right, I was actually thinking about that. Reason that I chose not to is that the absence of setting the variable evaluates to false and it was quicker like that to implement the code. For eg. https://github.com/elastic/beats/blob/feef1138f809281ac0a70ec97be6ed782901c2ab/metricbeat/module/kubernetes/util/kubernetes.go#L1231 the absence here would mean that this will evaluate to false and all will work. 

This would mean that in beats we will need to revert the logic. 
In a nutshell, 
So not setting use_kubeadm -> use_kubeadm: false in beats module but in here KubeAdm should be true

Comment (gizas) [reply kubernetes/metadata/config.go]: @tetianakravchenko I repeated the tests with use_kubeadm 

Final PR has the logic reverted , I mean that use_kubeadm is true by default
Users need to disable (use_kubeadm: false) to see no api calls

I have run tests with metricbeat:

Initial run is with metricbeat-8.15.0-SNAPSHOT: 170 api calls
Second run is with new build of metricbeat and not setting the use_kubeadm variable (so is true): 170 api calls
3rd run is with new build of metricbeat and setting use_kubeadm variable=false: 0 api calls


Screenshot 2024-07-19 at 1 00 22 PM

Reverting back returns the calls

**Review by tetianakravchenko — APPROVED** [review]

File: kubernetes/metadata/config.go (hunk 2 above)
Comment (tetianakravchenko) [reply kubernetes/metadata/config.go]: Thank you for reverting the logic and tests!

**Review by cmacknz — APPROVED** [review]

**Comment by ycombinator** [comment]
(Note to self and anyone else watching this PR) @gizas is on PTO until the last week of August.
